contrib/checkbox-ce-oem/checkbox-provider-ce-oem/bin/camera_utils.py

```diff
@@ -0,0 +1,930 @@
+                "No handler method found for scenario '{}'".format(
+                    scenario_name
+                )
+            )
+
+        if not isinstance(scenario_data, list):
+            raise ValueError(
+                "Scenario data for '{}' must be a list".format(scenario_name)
+            )
+
+        handler_method(scenario_data)
+
+    def _validate_scenario_item(
+        self, item: dict, required_fields: list
+    ) -> None:
+        """
+        Validate that a scenario item contains all required fields.
+        """
+        missing_fields = [
+            field for field in required_fields if field not in item
+        ]
+        if missing_fields:
+            log_and_raise_error(
+                "Missing required fields in scenario item: {}".format(
+                    missing_fields
+                ),
+                CameraConfigurationError,
+            )
+
+    def _process_scenario_items(
+        self, scenarios: list, scenario_type: str
+    ) -> None:
+        """
+        Base function to process scenario items and generate resources.
+
+        Args:
+            scenarios: List of scenario items to process
+            scenario_type: Type of scenario
+                (e.g., 'capture_image', 'record_video')
+
+        Raises:
+            CameraConfigurationError: If scenario item is invalid
+            Exception: If error occurs during processing
+        """
+        required_fields = [
+            "camera",
+            "method",
+            "physical_interface",
+            "v4l2_device_name",
+            "resolutions",
+            "formats",
+        ]
+
+        for item in scenarios:
+            try:
+                self._validate_scenario_item(item, required_fields)
+
+                for r, f in product(item["resolutions"], item["formats"]):
+                    resource_item = {
+                        "scenario": self._current_scenario_name,
+                        "camera": item["camera"],
+                        "method": item["method"],
+                        "physical_interface": item["physical_interface"],
+                        "v4l2_device_name": item["v4l2_device_name"],
+                        "format": f,
+                        "width": r["width"],
+                        "height": r["height"],
+                    }
+
+                    # Add scenario-specific fields
+                    if scenario_type == "record_video":
+                        resource_item["fps"] = r["fps"]
+
+                    self._resource_items.append(resource_item)
+
+            except Exception as e:
+                logger.error(
+                    "Error processing {} item: {}".format(scenario_type, e)
+                )
+                continue
+
+    def capture_image(self, scenarios: list) -> None:
+        """
+        Handle and generate the resource of the capture_image scenario
+        """
+        self._process_scenario_items(scenarios, "capture_image")
```

Comment:
Leverage the CameraScenarios Enum

```suggestion
        self._process_scenario_items(scenarios, CameraScenarios.CAPTURE_IMAGE.value)
```